packages/bolt-connection/test/bolt/bolt-protocol-v1.test.js

```diff
@@ -334,6 +334,59 @@ describe('#unit BoltProtocolV1', () => {
     })
   })
 
+  describe('Bolt v5.1', () => {
+    /**
+     * @param {string[]} notificationFilters The impersonated user.
+     * @param {function(protocol: BoltProtocolV1)} fn
+     */
+    function verifyNotificationFiltersNotSupportedError (notificationFilters, fn) {
+      const recorder = new utils.MessageRecordingConnection()
+      const protocol = new BoltProtocolV1(recorder, null, false, undefined, undefined, () => {})
+
+      expect(() => fn(protocol)).toThrowError(
+        'Driver is connected to the database that does not support user notification filters. ' +
+        'Please upgrade to neo4j 5.3.0 or later in order to use this functionality. ' +
+        `Trying to set notifications to ${JSON.stringify(notificationFilters)}.`
+      )
+    }
+
+    describe('initialize', () => {
+      function verifyInitialize (notificationFilters) {
+        verifyNotificationFiltersNotSupportedError(
+          notificationFilters,
+          protocol => protocol.initialize({ notificationFilters }))
+      }
+
+      it('should throw error when notificationFilters is set', () => {
+        verifyInitialize('test')
```

Comment:
Am I misreading the code or should this be
```suggestion
        verifyInitialize(['test'])
```
(more places, won't comment)